master/buildbot/buildbot_net_usage_data.py

```diff
@@ -143,14 +147,45 @@ def computeUsageData(master):
     return data
 
 
-def _sendBuildbotNetUsageData(data):
-    log.msg("buildbotNetUsageData: sending {}".format(data))
+def _sendWithUrlib(url, data):
     data = json.dumps(data)
     clen = len(data)
-    req = urllib2.Request(PHONE_HOME_URL, data, {'Content-Type': 'application/json', 'Content-Length': clen})
-    f = urllib2.urlopen(req)
+    req = urllib_request.Request(url, data, {
+        'Content-Type': 'application/json',
+        'Content-Length': clen
+    })
+    try:
+        f = urllib_request.urlopen(req)
+    except urllib.error.URLError:
+        return None
     res = f.read()
     f.close()
+    return res
+
+
+def _sendWithRequests(url, data):
+    try:
+        import requests
+    except ImportError:
+        return None
+    r = requests.post(url, json=data)
+    return r.text
+
+
+def _sendBuildbotNetUsageData(data):
+    log.msg("buildbotNetUsageData: sending {}".format(data))
+    # first try with requests, as this is the most stable http library
+    res = _sendWithRequests(PHONE_HOME_URL, data)
+    # then we try with stdlib, which not always work with https
+    if res is None:
+        res = _sendWithUrlib(PHONE_HOME_URL, data)
+    # at last stage
+    if res is None:
+        log.msg("buildbotNetUsageData: Could not send using https, "
+                "please `pip install 'requests[security]'` for proper SSL implementation`")
+        data['buggySSL'] = True
+        res = _sendWithUrlib2(PHONE_HOME_URL.replace("https://", "http://"), data)
```

Comment:
This should be _sendWithUrlilb()

Can you fix this and then get this merged in?  This will fix a bunch of tests on Python 3.